### ISSUE TITLE
Support disabling automatic BIOS fan control

### ISSUE THREAD

**Issue by giomasce** [vitorafsr/i8kutils#26]
Hi. Starting from version 5.5 Linux supports disabling the automatic BIOS fan control through the `sysfs` interface of module `dell-smm-hwmon`. It would be nice if `i8kutils` detected this possibility and automatically disabled BIOS fan control when applicable. See `pwm1_enable` in [the relevant documentation](https://www.kernel.org/doc/html/latest/hwmon/dell-smm-hwmon.html).

Notice that for the moment very few laptop models actually support this feature, because Linux maintainers want to make it available only on models [where it has been tested](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/hwmon/dell-smm-hwmon.c#n1203). The documentation page linked above also details how to test your own model and possibly report a patch to make it work.

**Comment by kylebakerio** [comment]
Yeah, this would be really nice to have. I've added a walkthrough on this answer for implementing it on a supported model: https://askubuntu.com/a/1319565/422405

but it would be ideal if it was built in.

**Comment by kylebakerio** [comment]
https://www.kernel.org/doc/html/latest/hwmon/dell-smm-hwmon.html

> If pwm1_enable is not available, then it means that SMM codes for enabling and disabling automatic BIOS fan control are not whitelisted for your hardware. It is possible that codes that work for other laptops actually work for yours as well, or that you have to discover new codes.

> Check the list i8k_whitelist_fan_control in file drivers/hwmon/dell-smm-hwmon.c in the kernel tree: as a first attempt you can try to add your machine and use an already-known code pair. If, after recompiling the kernel, you see that pwm1_enable is present and works (i.e., you can manually control the fan speed), then please submit your finding as a kernel patch, so that other users can benefit from it. Please see Submitting patches: the essential guide to getting your code into the kernel for information on submitting patches.
